### ISSUE TITLE
`Mutable` doesn't work for constants

### ISSUE THREAD

**Issue by WilsonGramer** [wipplelang/wipple#68]
The following code prints `42` twice, instead of `42` followed by `43`:

```wipple
m :: Mutable Number
m : mutable 42

show (get m)
increment! m
show (get m)
```

Removing the first line to make `m` a variable, however, causes the code to behave as expected.

**Comment by WilsonGramer** [comment]
The issue is that constants are monomorphized every time they are used, so `m` refers to a different constant each time it's used. Instead, we should only generate different constants when the types are different.

**Comment by WilsonGramer** [comment]
Will be resolved by 4ad409faf823a2e6fda0bbf00ceedf9fe138ebde!
